src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,435 @@
+
+    @ArgumentCollection
+    private final GATKAnnotationArgumentCollection userArgs;
+
+    // Map of Annotation (simple) class names to the corresponding discovered plugin instance
+    private final Map<String, Annotation> allDiscoveredAnnotations = new HashMap<>();
+
+    // Map of Annotation (simple) class names to the corresponding default plugin instance
+    // it is a LinkedHashMap because we want to remember the order in which these were provided, and also keep the
+    // actual instances in case they have any additional state provided by the tool
+    // when they were created
+    private final Map<String, Annotation> toolDefaultAnnotations = new LinkedHashMap<>();
+    private final Set<String> toolDefaultGroups = new HashSet<>();
+
+    // Set of predecessor annotations for which we've seen arguments exist either as a tool default or be supplied by the user
+    // (eg. InbreedingCoeff if we see "--founderID" on the command line)
+    private final Set<String> requiredPredecessors = new HashSet<>();
+
+    // Map of annotation group name to list of annotations flagged with that group. The key here the simple name of the interface
+    // that describes the annotation group.
+    private final Map<String, List<Annotation>> discoveredGroups = new HashMap<>();
+
+    /**
+     * @return the class object for the base class of all plugins managed by this descriptor
+     */
+    @Override
+    public Class<?> getPluginClass() {
+        return pluginBaseClass;
+    }
+
+    /**
+     * A list of package names which will be searched for plugins managed by the descriptor.
+     *
+     * @return Strings of package names over which the plugin descriptor will search for Annotation Classes
+     */
+    @Override
+    public List<String> getPackageNames() {
+        return Collections.singletonList(pluginPackageName);
+    }
+
+    /**
+     * Constructor that allows client tools to specify what annotations (optionally with parameters specified) to use as their defaults
+     * before discovery of user specified annotations.
+     *
+     * @param userArgs           Argument collection to control the exposure of the command line arguments.
+     * @param toolDefaultAnnotations Default annotations that may be supplied with arguments
+     *                               on the command line. May be null.
+     * @param toolDefaultGroups List of tool specified default annotation group names. Annotations specified this way
+     *                          will be instantiated with default arguments. may be null.
+     */
+    public GATKAnnotationPluginDescriptor(final GATKAnnotationArgumentCollection userArgs, final List<Annotation> toolDefaultAnnotations, final List<Class<? extends Annotation>> toolDefaultGroups) {
+        this.userArgs = userArgs;
+        if (null != toolDefaultAnnotations) {
+            toolDefaultAnnotations.forEach(f -> {
+                final Class<? extends Annotation> rfClass = f.getClass();
```

Comment:
There are a few "rf" prefixed variables throughout this file that should probably be renamed.

Reply:
done